src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest.java

```diff
@@ -36,102 +37,56 @@
  */
 public class AnalyzeRequest extends SingleCustomOperationRequest<AnalyzeRequest> {
 
-    private String text;
-
-    private String analyzer;
-
-    private String tokenizer;
-
-    private String[] tokenFilters = Strings.EMPTY_ARRAY;
-
-    private String[] charFilters = Strings.EMPTY_ARRAY;
-
-    private String field;
+    private BytesReference source;
+    private boolean unsafe;
```

Comment:
I think we can. The reusing of the request body has always been there and it used in other apis too, maybe instead we should open a different issue and discuss if we should drop this? This would simplify the code and remove a potential pitfall (the unsafe support isn't a straightforward pattern)